pkg/plugin/v2/plugin.go

```diff
@@ -35,15 +35,29 @@ var (
 	_ plugin.CreateWebhookPluginGetter = Plugin{}
 )
 
+// Plugin defines the Plugin. e.g go.kubebuilder.io/v3-alpha
 type Plugin struct {
 	initPlugin
 	createAPIPlugin
 	createWebhookPlugin
 }
 
-func (Plugin) Name() string                                   { return pluginName }
-func (Plugin) Version() plugin.Version                        { return pluginVersion }
-func (Plugin) SupportedProjectVersions() []string             { return supportedProjectVersions }
-func (p Plugin) GetInitPlugin() plugin.Init                   { return &p.initPlugin }
-func (p Plugin) GetCreateAPIPlugin() plugin.CreateAPI         { return &p.createAPIPlugin }
+// Name returns the name of the plugin. e.g go.kubebuilder.io
```

Comment:
I don't think comments are required on concrete methods of an interface, and if they exist
can be more confusing/duplicative. I'd prefer these removed if they are not required to pass the linter.

Reply:
Developers are able to use the interface to create their own plugins as use the plugins implementations. In this way, IHMO both needs to be properly documented. However, I approved it as well based in your comment.

Reply:
Developers are able to use the interface to create their own plugins as use the plugins implementations. In this way, IHMO both needs to be properly documented. However, I approved it as well based in your comment.

Reply:
The interface already describes the method's contract. If there are any special things a particular concrete method does, they can be added in a comment; none of the comments being added give any extra relevant information so they aren't helpful and can drift easily. I'd prefer if you removed them, but I won't block the PR on this.